scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -440,32 +505,225 @@ static void refresh_tune_params(void)
 	}
 }
 
+/*
+ * log2 helper functions taken from scx_lavd
+ */
+static inline __attribute__((always_inline)) u32 bpf_log2(u32 v)
+{
+        u32 r;
+        u32 shift;
+
+        r = (v > 0xFFFF) << 4; v >>= r;
+        shift = (v > 0xFF) << 3; v >>= shift; r |= shift;
+        shift = (v > 0xF) << 2; v >>= shift; r |= shift;
+        shift = (v > 0x3) << 1; v >>= shift; r |= shift;
+        r |= (v >> 1);
+        return r;
+}
+
+static inline __attribute__((always_inline)) u32 bpf_log2l(u64 v)
+{
+        u32 hi = v >> 32;
+        if (hi)
+                return bpf_log2(hi) + 32 + 1;
+        else
+                return bpf_log2(v) + 1;
+}
+
+static u64 max_clamp(u64 val, u64 max)
+{
+	return val <= max ? val : max;
+}
+
+/*
+ * ** Taken directly from fair.c in the Linux kernel **
+ *
+ * We use this table to inversely scale deadline according to a task's
+ * calculated latency factor. We preserve the comment directly from the table
+ * in fair.c:
+ *
+ * "Nice levels are multiplicative, with a gentle 10% change for every
+ * nice level changed. I.e. when a CPU-bound task goes from nice 0 to
+ * nice 1, it will get ~10% less CPU time than another CPU-bound task
+ * that remained on nice 0.
+ *
+ * The "10% effect" is relative and cumulative: from _any_ nice level,
+ * if you go up 1 level, it's -10% CPU usage, if you go down 1 level
+ * it's +10% CPU usage. (to achieve that we use a multiplier of 1.25.
+ * If a task goes up by ~10% and another task goes down by ~10% then
+ * the relative distance between them is ~25%.)"
+ */
+const int sched_prio_to_weight[DL_MAX_LAT_PRIO] = {
+ /* -20 */     88761,     71755,     56483,     46273,     36291,
+ /* -15 */     29154,     23254,     18705,     14949,     11916,
+ /* -10 */      9548,      7620,      6100,      4904,      3906,
+ /*  -5 */      3121,      2501,      1991,      1586,      1277,
+ /*   0 */      1024,       820,       655,       526,       423,
+ /*   5 */       335,       272,       215,       172,       137,
+ /*  10 */       110,        87,        70,        56,        45,
+ /*  15 */        36,        29,        23,        18,        15,
+};
+
+static u64 sched_prio_to_latency_weight(u64 prio)
+{
+	if (prio >= DL_MAX_LAT_PRIO) {
+		scx_bpf_error("Invalid prio index");
+		return 0;
+	}
+
+	return sched_prio_to_weight[DL_MAX_LAT_PRIO - prio - 1];
```

Comment:
So, this can just return `p->scx.weight` if the rest of the code assumes [1, 10000] scale.